Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Paged/EthereumProviderScriptHandler.swift

```diff
@@ -99,13 +99,16 @@ class EthereumProviderScriptHandler: TabContentScript {
       firstAllowedAccount: String,
       updateJSProperties: Bool
     ) {
-      if reject {
-        replyHandler(nil, formedResponse.jsonString)
-      } else {
-        replyHandler(formedResponse.jsonObject, nil)
-      }
-      if updateJSProperties {
-        tab.updateEthereumProperties()
+      Task { @MainActor in
```

Comment:
If `replyHandler` doesn't need to be executed on main then this is not required

Reply:
Fixed.